db/dump.py

```diff
@@ -555,3 +684,101 @@ def _get_incremental_dump_timestamp(dump_id=None):
+    """
+    archive_name = os.path.basename(archive_path).split('.')[0]
+    with open(archive_path, "w") as archive:
+
+        pxz_command = ["pxz", "--compress"]
+        if threads is not None:
+            pxz_command.append("-T %s" % threads)
+        pxz = subprocess.Popen(pxz_command, stdin=subprocess.PIPE, stdout=archive)
+
+        # Creating the archive
+        with tarfile.open(fileobj=pxz.stdin, mode="w|") as tar:
+            # TODO: Get rid of temporary directories and write directly to tar file if that's possible
+            temp_dir = tempfile.mkdtemp()
+
+            # Adding metadata
+            schema_seq_path = os.path.join(temp_dir, "SCHEMA_SEQUENCE")
+            with open(schema_seq_path, "w") as f:
+                f.write(str(db.SCHEMA_VERSION))
+            tar.add(schema_seq_path,
+                    arcname=os.path.join(archive_name, "SCHEMA_SEQUENCE"))
+            timestamp_path = os.path.join(temp_dir, "TIMESTAMP")
+            with open(timestamp_path, "w") as f:
+                f.write(time_now.isoformat(" "))
+            tar.add(timestamp_path,
+                    arcname=os.path.join(archive_name, "TIMESTAMP"))
+            tar.add(DUMP_LICENSE_FILE_PATH,
+                    arcname=os.path.join(archive_name, "COPYING"))
+
+            archive_tables_dir = os.path.join(temp_dir, "abdump", "abdump")
+            utils.path.create_path(archive_tables_dir)
+            _copy_tables(archive_tables_dir, dataset_dump, tar, archive_name, start_t, end_t)
+            tar.add(archive_tables_dir, arcname=os.path.join(archive_name, "abdump"))
+
+            shutil.rmtree(temp_dir)
+
+        pxz.stdin.close()
+        pxz.wait()
+
+
+def dump_dataset_tables(location, threads=None):
+    """Create full dump of dataset tables in a specified location.
+
+    Args:
+        location: Directory where archive will be created.
+    Returns:
+        Path to created dump.
+    """
+    utils.path.create_path(location)
+    time_now = datetime.today()
+    archive_name = "acousticbrainz-dataset-dump-%s" % time_now.strftime("%Y%m%d-%H%M%S")
+
+    archive_path = os.path.join(location, archive_name + ".tar.xz")
+    _dump_tables(
+        archive_path=archive_path,
+        threads=threads,
+        dataset_dump=True,
+        time_now=time_now,
+    )
+    return archive_path
+
+
+def import_datasets_dump(archive_path):
+    """Import datasets from .tar.xz archive into the database."""
+    archive_name = os.path.basename(archive_path).split('.')[0]
+    pxz_command = ["pxz", "--decompress", "--stdout", archive_path]
+    pxz = subprocess.Popen(pxz_command, stdout=subprocess.PIPE)
+
+    table_names = _DATASET_TABLES.keys()
+    connection = db.engine.raw_connection()
+    try:
+        cursor = connection.cursor()
+        with tarfile.open(fileobj=pxz.stdout, mode="r|") as tar:
+            temp_dir = tempfile.mkdtemp()
+            tar.extractall(temp_dir)
```

Comment:
This is different to the data file dumps, where we stream the data out a file at a time instead of all at once. Is there a reason for the difference?

Reply:
addressed in #238